### ISSUE TITLE
Remove silent overwriting of `shutil.copy` in ARMI utils

### ISSUE THREAD

**Issue by opotowsky** [terrapower/armi#1690]
`safeCopy` silently overwrites `shutil.copy` in ARMI:

https://github.com/terrapower/armi/blob/aa758f3a49d9e71f0ba4d616fcaf49df4091bfa5/armi/utils/__init__.py#L845-L847

I'm making this ticket so that when `shutil.copy` is replaced in all downstream plugins with `safeCopy`, I want to remove this menace from the code base. 

I also need to check in with other ARMI users so they have a heads up on a potentially breaking change (so, if shutil is being leveraged, they can do nothing and revert to the real `shutil.copy`, or they can also swap to explicitly using `safeCopy`)

**Comment by john-science** [comment]
I agree, silently over-writing a standard library tool whenever you `import armi` is a bad solution to whatever problem this was trying to solve.

I would say "let's do it right now!", but since it's an API-breaking change, we will be slow and deliberate.

@drewj-usnctech Just FYI.

**Comment by onufer** [comment]
this seems super low priority. I agree it is confusing and surprising but its going to be a lot of work to change and very little benifit.

**Comment by opotowsky** [comment]
Thanks for reopening @onufer 

I grepped around and it won't be a huge lift. I agree it's low priority, but I'd like to get in the habit of paying back some tech debt and this is an easy one.

**Comment by john-science** [comment]
> this seems super low priority.

I can see that from a "fixing this doesn't help us model a reactor" stand point. I really can.

But, from a quality perspective, this is a nasty piece of ugly spaghetti code that makes us look unprofessional and sloppy.

Thus my concern.

Also, I could fix this problem in like... 2 minutes in each downstream repo that is affected. So, it's just the easiest thing in the world to fix.

**Comment by opotowsky** [comment]
Agreed. It actually can be done piecemeal, and doesn't need massive coordination.

1. replace `shutil.copy` with `safeCopy` in ARMI
2. repeat one by one for each plugin
3. last, remove the `shutil.copy` overwrite in ARMI.
